pkg/reconciler/reconciler.go

```diff
@@ -534,15 +547,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 	return ctrl.Result{RequeueAfter: r.reconcilePeriod}, nil
 }
 
-func (r *Reconciler) getValues(obj *unstructured.Unstructured) (chartutil.Values, error) {
-	crVals, err := internalvalues.FromUnstructured(obj)
+func (r *Reconciler) getValues(ctx context.Context, obj *unstructured.Unstructured) (chartutil.Values, error) {
+	vals, err := r.valueTranslator.Translate(ctx, obj)
 	if err != nil {
 		return chartutil.Values{}, err
 	}
+	crVals := internalvalues.New(vals)
 	if err := crVals.ApplyOverrides(r.overrideValues); err != nil {
 		return chartutil.Values{}, err
 	}
-	vals := r.valueMapper.Map(crVals.Map())
+	vals = r.valueMapper.Map(crVals.Map())
```

Comment:
agreed, having `valueMapper` is not of much use here. Instead of using both `valueTranslator` and `valueMapper` on the same values can we prioritize the translatorfunc if its not nil by the user?

Reply:
@varshaprasad96 can you please clarify what you mean by "prioritize the translatorfunc if its not nil by the user"?
Do you mean passing both `WithValueMapper` and `WithValueTranslator` to `New()` should be rejected?

Reply:
If `Translator` is passed, we can ignore the mapper (in case both are non-nil). This would make it easier to deprecate and remove mapper completely later. Thoughts?

Reply:
I'm afraid it would be detrimental to developer experience if the `WithValueMapper` option was silently ignored when the `WithValueTranslator` option would be passed. IMHO it would violate the principle of least astonishment.

Reply:
Completely agree. I had logging a warning and docs along with it in mind so that we don't have to make a breaking change of removing `WithValueMapper` immediately. We can probably bring this up and discuss in tomorrow's [community meeting](https://github.com/operator-framework/community#helm-operator-plugins-working-group). It would really be nice to get this PR merged as it's a really helpful addition :)

Reply:
Unfortunately I'll be in a place with poor network coverage at that time so I will not be able to join the meeting. But to summarise my thoughts:
- I don't think we need to remove value mappers in this PR, or even the same release, but it would make sense to add deprecation warnings to prepare for removal if we want to do it eventually,
- a warning and docs alone are in my experience insufficient countermeasure to a surprise in the API unless there is no other option, and I think we have a good alternative:
- IF we want to make mappers and translators exclusive, then I'd strongly recommend returning and error from New in case both `WithValueMapper` and `WithValueTranslator` are passed. This is not going to harm existing users, unless they try using a translator, in which case such error is a useful hint.

Reply:
Hi, any updates on this after the meeting? How do we proceed on this?

Also, would it be possible to add me to the `operator-framework` org? Just so that I can configure GitHub to send PR notifications to my work email..